grails-app/services/UserProfileService.groovy

```diff
@@ -0,0 +1,29 @@
+import grails.transaction.Transactional
+import org.springframework.security.core.authority.SimpleGrantedAuthority
+import org.transmart.searchapp.AuthUser
+import grails.converters.JSON
+
+@Transactional
```

Comment:
remove this if it's not really transactional and annotate individual methods that write to the database